### DISCUSSION TITLE
`todo-comments` unable to load highlight group colors after `auto-session` restoration

### DISCUSSION THREAD

**Discussion by teoshibin** [NvChad/NvChad#2712] [Q&A] [answered]
Long story short, `todo-comments` highlighting won't get the correct color of `base46` after session restore of `auto-session`, but other colors in the editor looks fine. I believe this has something todo with how cached highlight groups are loaded, maybe the order it loads etc. Not a lua expert here so I have no idea how to properly fix it. I just do a very lazy and straight forward workaround to fix it, documented below.

I don't even know if this is even a bug so I'll put it here in the discussion.

## Reproducing

So first we'll need 2 external plugins `todo-comments` and `auto-session`. `base46` is installed.
plugin config for `auto-session`
```lua
  {
    -- open up the same stuff after closing editor
    "rmagatti/auto-session",
    lazy = false,
    config = function()

      -- NOTE: fix nvimtree buffer layout
      -- https://github.com/NvChad/ui/issues/132
      vim.api.nvim_create_autocmd({ "BufEnter" }, {
        pattern = "NvimTree*",
        callback = function()
          local api = require "nvim-tree.api"
          local view = require "nvim-tree.view"
          if not view.is_visible() then
            api.tree.open()
          end
        end,
      })

      require("auto-session").setup {
        auto_session_suppress_dirs = { "~/", "~/Projects", "~/Downloads", "/" },
      }
    end,
  },
```
plugin config for todo comments
```lua
  {
    "folke/todo-comments.nvim",
    dependencies = { "nvim-lua/plenary.nvim" },
    opts = {},
    event = "BufReadPost",
  },
```
now we could do 
```txt
:SessionSave
:wqa
nvim
```
and that will open up the editor with the previously saved session
saving it isn't mandatory but we want to be sure it is saved just for demonstration.

## Result
Now, the problem arise, the following is what should've been displayed
![image](https://github.com/NvChad/NvChad/assets/62151367/23f1e9f6-67f1-4f2e-93d9-21d0417cf0db)
And what is actually displayed after session restoration is as follows
![image](https://github.com/NvChad/NvChad/assets/62151367/58acb3a3-1b7d-4a59-9a97-96b0ab9f72a9)
I also tried the following command after restoration, the color is correct which is even more confusing why it didn't work for `todo-comments`.
![image](https://github.com/NvChad/NvChad/assets/62151367/dbd4ad4c-3e63-4f43-a1b8-394f8545a33b)

## Extra
Now looking into their `todo-comments` source code, we see some configs stuff related to highlight groups and some fallback colors (I believe the white we're getting is simply because there's no fallback for it, but the rest will just fallback to their hardcoded colors)
- [their color config](https://github.com/folke/todo-comments.nvim/blob/833d8dd8b07eeda37a09e99460f72a02616935cb/lua/todo-comments/config.lua#L59)
- [where they do the highlighting](https://github.com/folke/todo-comments.nvim/blob/833d8dd8b07eeda37a09e99460f72a02616935cb/lua/todo-comments/highlight.lua#L205)   

After some more digging around I found out that this is assigned to build, which is the point I realized something is cached.
```lua
  -- plugin/init.lua
  {
    "NvChad/base46",
    branch = "v2.0",
    build = function()
      require("base46").load_all_highlights()
    end,
  },
```

## Workaround

As I can't be bothered diving any deeper, I just call load all highlights again for the plugin, but this defeats the purpose of the cache.
```lua
  {
    "folke/todo-comments.nvim",
    dependencies = { "nvim-lua/plenary.nvim" },
    config = function()
      require("base46").load_all_highlights()  -- here
      require("todo-comments").setup({
        signs = false,
      })
    end,
    event = "BufReadPost",
  },
```
I am more than happy to discuss about this if anyone can find out why this happens.

**Comment by teoshibin** [comment, marked as answer]
Ah v2.5 contains Todo highlight groups integration that can be loaded. Nice

**Reply by siduck** [reply]
@teoshibin  i didnt comment here  as i was releasing v2.5 which would also fix your issue :grimacing:
